src/lib/operators/projection.cpp

```diff
@@ -97,47 +97,109 @@ std::shared_ptr<const Table> Projection::_on_execute() {
-        column_is_nullable[column_id] =
-            column_is_nullable[column_id] || input_table.column_is_nullable(pqp_column_expression.column_id);
-        forwarding_cost += timer.lap();
-      } else {
-        // Newly generated column - the expression needs to be evaluated
-        auto output_segment = evaluator.evaluate_expression_to_segment(*expression);
-        column_is_nullable[column_id] = column_is_nullable[column_id] || output_segment->is_nullable();
-
-        // Storing the result in output_segments means that the vector may contain both ReferenceSegments and
-        // ValueSegments. We deal with this later.
-        output_segments[column_id] = std::move(output_segment);
-        expression_evaluator_cost += timer.lap();
+        column_is_nullable_by_chunk[chunk_id][column_id] =
+          column_is_nullable_by_chunk[chunk_id][column_id] ||
+          input_table.column_is_nullable(pqp_column_expression.column_id);
       }
     }
+    forwarding_cost += timer.lap();
 
+    // Output segments are associated with their chunk. At the moment the segments are only ReferenceSegments.
     output_segments_by_chunk[chunk_id] = std::move(output_segments);
+
+    // All columns are forwarded. We do not need to evaluate newly generated columns.
+    if (forwared_count == expression_count) continue;
+
+    // Defines the job that performs the evaluation if the columns are newly generated.
+    auto perform_projection_evaluation = [
+      this,
+      chunk_id,
+      &uncorrelated_subquery_results,
+      expression_count,
+      &output_segments_by_chunk,
+      &column_is_nullable_by_chunk
+      ] () {
+        ExpressionEvaluator evaluator(left_input_table(), chunk_id, uncorrelated_subquery_results);
+
+        for (auto column_id = ColumnID{0}; column_id < expression_count; ++column_id) {
+          const auto& expression = expressions[column_id];
+          if (expression->type != ExpressionType::PQPColumn) {
+            // Newly generated column - the expression needs to be evaluated
+            auto output_segment = evaluator.evaluate_expression_to_segment(*expression);
+            column_is_nullable_by_chunk[chunk_id][column_id] =
+              column_is_nullable_by_chunk[chunk_id][column_id] ||
+              output_segment->is_nullable();
+            // Storing the result in output_segments_by_chunk means that the vector
+            // for the separate chunks may contain both ReferenceSegments and
+            // ValueSegments. We deal with this later.
+            output_segments_by_chunk[chunk_id][column_id] = std::move(output_segment);
+          }
+        }
+    };
+    // It is checked if the job should be executed in parallel. If not, it will be executed right away.
+    // The upper bound of the chunk size,  which defines if it will be executed in parallel or not,
+    // still needs to be re-evaluated over time to find the value which gives the best performance.
```

Comment:
Suggestion for the first sentence:
> Evaluate the expression immediately if it contains less than `job_spawn_threshold` rows, otherwise wrap it into a task

Also, there is a superflous space in the second line.